packages/ckeditor5-link/src/autolink.ts

```diff
@@ -104,6 +107,54 @@ export default class AutoLink extends Plugin {
 	public afterInit(): void {
 		this._enableEnterHandling();
 		this._enableShiftEnterHandling();
+		this._enablePasteLinking();
+	}
+
+	/**
+	 * Enables autolinking on pasting a URL.
+	 */
+	private _enablePasteLinking(): void {
+		const editor = this.editor;
+		const model = editor.model;
+		const selection = model.document.selection;
+		const clipboardPipeline = editor.plugins.get( 'ClipboardPipeline' );
+		const linkCommand = editor.commands.get( 'link' )!;
+
+		const updateLink = ( newLink: string ): boolean => {
+			const selStart = selection.getFirstPosition()!;
+			const selEnd = selection.getLastPosition()!;
+
+			const linkRange = findAttributeRange( selStart, 'linkHref', selection.getAttribute( 'linkHref' ), model );
+			// why is containsPosition false when the point is equal to the start or end of the range?
+			if ( linkRange.containsPosition( selEnd ) || linkRange.end.isEqual( selEnd ) ) {
+				// update the whole link if one is partially selected
+				model.change( writer => writer.setAttribute( 'linkHref', newLink, linkRange ) );
+				return true;
+			} else {
+				return false;
+			}
+		};
+
+		clipboardPipeline.on( 'inputTransformation', ( evt, data: ClipboardInputTransformationData ) => {
+			if ( linkCommand.isEnabled && !selection.isCollapsed ) {
```

Comment:
ah, I hate early returns so that'll be fun 😂 

I had a check for `this.isEnabled` at one point but it must've been lost in other changes I was making. I'll put it back.